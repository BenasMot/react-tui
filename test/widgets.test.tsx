import React from "react";
import { getScreen } from "./fixtures/screen";
import { screenToString } from "./util/screen";
import { createTestRenderer, settle } from "./fixtures/render";
import { RootBox } from "./fixtures/components/RootBox";

it("<box />", async () => {
  const screen = getScreen();
  const renderer = createTestRenderer(screen);
  renderer.render(
    <RootBox>
      <box
        label="child-box"
        border={{ type: "line" }}
        style={{ border: { fg: "blue" } }}
      >
        best-box
      </box>
    </RootBox>
  );
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
});

it("<list />", async () => {
  const screen = getScreen();
  const renderer = createTestRenderer(screen);
  renderer.render(
    <RootBox>
      <list
        label="child-list"
        border={{ type: "line" }}
        style={{ border: { fg: "blue" } }}
        items={["a", "b", "c", "1", "2", "3"]}
      ></list>
    </RootBox>
  );
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
});

it("<list />, long", async () => {
  const screen = getScreen();
  const renderer = createTestRenderer(screen);
  renderer.render(
    <RootBox>
      <list
        label="long-list"
        border={{ type: "line" }}
        style={{ border: { fg: "blue" } }}
        items={" "
          .repeat(30)
          .split("")
          .map((_, i) => `${i}`)}
      ></list>
    </RootBox>
  );
  await settle();
  expect(screenToString(screen)).toMatchSnapshot();
});
