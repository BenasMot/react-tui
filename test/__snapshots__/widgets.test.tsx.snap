// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<box /> 1`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─child-box──────────────────────────────────────────────┐│
││best-box                                                ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;

exports[`<list /> 1`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─child-list─────────────────────────────────────────────┐│
││a                                                       ││
││b                                                       ││
││c                                                       ││
││1                                                       ││
││2                                                       ││
││3                                                       ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;

exports[`<list />, long 1`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─long-list──────────────────────────────────────────────┐│
││0                                                       ││
││1                                                       ││
││2                                                       ││
││3                                                       ││
││4                                                       ││
││5                                                       ││
││6                                                       ││
││7                                                       ││
││8                                                       ││
││9                                                       ││
││10                                                      ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;
